[reactive-element] Type 'Component' is not assignable to type 'ReactiveElement'. Types have separate declarations of a private property '__instanceProperties'.

### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

With lit@2.2.8 lit decorators works as expected, but if I wanted to upgrade to 2.3.0 or above I get this typescript error:

```
component.ts: error TS2345: Argument of type 'Component' is not assignable to parameter of type 'ReactiveElement | ClassElement'.
  Type 'Component' is not assignable to type 'ReactiveElement'.
    Types have separate declarations of a private property '__instanceProperties'.
```

Component code:
```
export class Component extends LitElement {
	@query('[part="feedback"]') readonly $feedback!: HTMLElement;
	...
}
```

![image](https://user-images.githubusercontent.com/8919920/186157901-4089aafe-712a-438d-bdb0-bff3043a2dae.png)


### Reproduction

I didn't had time to create a reproduction repo, but if it's needed I can make.

We use the following tsconfig.json compilerOptions:
```
{
  "compilerOptions": {
    "baseUrl": ".",
    "target": "es2018",
    "module": "esnext",
    "moduleResolution": "node",
    "declaration": true,
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "sourceMap": true,
    "resolveJsonModule": true,
    "useDefineForClassFields": false,
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.d.ts"
  ]
}

```

### Workaround

Workaround is to use 2.2.x version

### Is this a regression?

Yes. This used to work, but now it doesn't.

### Affected versions

Failing since 2.3.0, worked in 2.2.8

### Browser/OS/Node environment

node: 16.15.1
typescript: 4.7.4
OS: macOS 12.5

--- a/.changeset/heavy-bugs-vanish.md
+++ b/.changeset/heavy-bugs-vanish.md
@@ -1,0 +1,6 @@
+---
+'@lit/reactive-element': patch
+'lit': patch
+---
+
+Allow decorators to accept `ReactiveElement` class from a different source.

--- a/packages/reactive-element/src/decorators/base.ts
+++ b/packages/reactive-element/src/decorators/base.ts
@@ -6,6 +6,10 @@
 
 import {ReactiveElement} from '../reactive-element.js';
 
+export type Interface<T> = {
+  [K in keyof T]: T[K];
+};
+
 export type Constructor<T> = {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   new (...args: any[]): T;
@@ -71,7 +75,7 @@ export const decorateProperty =
     descriptor?: (property: PropertyKey) => PropertyDescriptor;
   }) =>
   (
-    protoOrDescriptor: ReactiveElement | ClassElement,
+    protoOrDescriptor: Interface<ReactiveElement> | ClassElement,
     name?: PropertyKey
     // Note TypeScript requires the return type to be `void|any`
     // eslint-disable-next-line @typescript-eslint/no-explicit-any

--- a/packages/reactive-element/src/decorators/base.ts
+++ b/packages/reactive-element/src/decorators/base.ts
@@ -6,6 +6,11 @@
 
 import {ReactiveElement} from '../reactive-element.js';
 
+/**
+ * Generates a public interface type that removes private and protected fields.
+ * This allows accepting otherwise compatible versions of the type (e.g. from
+ * multiple copies of the same package in `node_modules`).
+ */
 export type Interface<T> = {
   [K in keyof T]: T[K];
 };

